org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JSONUtility.java

```diff
@@ -61,6 +62,9 @@ private static <T> T toModel(Gson gson, Object object, Class<T> clazz) {
 		if (object instanceof String json) {
 			return gson.fromJson(json, clazz);
 		}
+		if (object instanceof Map map) {
+			return gson.fromJson(gson.toJsonTree(map), clazz);
+		}
```

Comment:
I don't think changes to this are necessary. I tried just a simple :

```
JSONUtility.toLsp4jModel("""
{
    "changes": {
        "file:///files/tests/java/45/src/main/java/App.java": [
            {
                "range": {
                    "start": {
                        "line": 0,
                        "character": 0
                    },
                    "end": {
                        "line": 0,
                        "character": 0
                    }
                },
                "newText": "import javax.swing.JFrame;"
            },
            {
                "range": {
                    "start": {
                        "line": 0,
                        "character": 0
                    },
                    "end": {
                        "line": 0,
                        "character": 0
                    }
                },
                "newText": "\n\n"
            }
        ]
    }
}""", WorkspaceEdit.class)
```

and it seemed to convert it as expected.